ui/src/app/tasks/task-definitions/task-definitions.component.ts

```diff
@@ -159,46 +159,46 @@ export class TaskDefinitionsComponent implements OnInit, OnDestroy {
       },
       {
         divider: true,
-        hidden: !this.authService.securityInfo.canAccess(['ROLE_CREATE'])
+        hidden: !this.authService.securityInfo.canAccess(['ROLE_DEPLOY'])
       },
       {
         id: 'launch-task' + index,
         icon: 'play',
         action: 'launch',
         title: 'Launch task',
         isDefault: true,
-        hidden: !this.authService.securityInfo.canAccess(['ROLE_CREATE'])
+        hidden: !this.authService.securityInfo.canAccess(['ROLE_DEPLOY'])
       },
       {
         divider: true,
-        hidden: !this.schedulerEnabled || !this.authService.securityInfo.canAccess(['ROLE_CREATE'])
+        hidden: !this.schedulerEnabled || !this.authService.securityInfo.canAccess(['ROLE_SCHEDULE'])
       },
       {
         id: 'task-schedule' + index,
         icon: 'clock-o',
         action: 'schedule',
         title: 'Schedule task',
         disabled: !this.schedulerEnabled,
-        hidden: !this.schedulerEnabled || !this.authService.securityInfo.canAccess(['ROLE_CREATE'])
+        hidden: !this.schedulerEnabled || !this.authService.securityInfo.canAccess(['ROLE_SCHEDULE'])
       },
       {
         id: 'delete-schedules' + index,
         icon: 'trash',
         action: 'delete-schedules',
         title: 'Delete schedule',
         disabled: !this.schedulerEnabled,
-        hidden: !this.schedulerEnabled || !this.authService.securityInfo.canAccess(['ROLE_CREATE'])
+        hidden: !this.schedulerEnabled || !this.authService.securityInfo.canAccess(['ROLE_SCHEDULE'])
       },
       {
         divider: true,
-        hidden: !this.schedulerEnabled || !this.authService.securityInfo.canAccess(['ROLE_CREATE'])
+        hidden: !this.schedulerEnabled || !this.authService.securityInfo.canAccess(['ROLE_DESTROY'])
```

Comment:
@oodamien Could you tell me the reason for this update? I'd like to understand how this part of the code works.